lib/pdn/pdn.c

```diff
@@ -643,8 +643,10 @@ int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
 		}
 	}
 	if (dns4_sec) {
-		if (zsock_inet_pton(AF_INET, dns4_sec_str, dns4_sec) != 1) {
```

Comment:
Could dns4_pri (and dns4_sec) be used to indicate that DNS address was not found, if it was asked? Setting the pointers to null is likely asking for trouble, but the bytes could be set to 0, to indicate no address. Another possibility would be to return how many addresses we found (or to add out parameters to indicate what we found).

Same for IPv6.

Reply:
Which one is logical? I can do any of these.

Reply:
Ok I made it with setting to 0. Now dns4_pri, dns4_sec, ipv4_mtu, dns6_pri, dns6_sec, ipv6_mtu will be containing zeros.

Reply:
This would at least allow the caller to check if the address was fetched successfully. However, I am starting to lean on having a separate function for getting MTU and for this to return the amount of DNS addresses that were found. But let's wait for libmodem comments before proceeding with any of this.